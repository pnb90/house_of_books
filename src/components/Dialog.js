import React, {useState} from 'react'
import Dialog from '@material-ui/core/Dialog'
import DialogContent from '@material-ui/core/DialogContent'
import DialogContentText from '@material-ui/core/DialogContentText'
import DialogTitle from '@material-ui/core/DialogTitle'
import styled from 'styled-components'

function DialogComponent(props) {
  
  const [open, setOpen] = useState(false)

  function handleClickOpen() {
    setOpen(true)
  }

  function handleClose() {
    setOpen(false)
  }

  const BookCover = styled.img`
    height: 266px;
    width: 200px;
    border: 1px solid grey;
    padding: 5px;    
  `

  const Row = styled.div`
    display: flex;
    align-items: center;
  `

  const CoverColumn = styled.div`
    flex: 25%;
  `
  
  const TitleColumn = styled.div`
    flex: 75%;
    flex-wrap: wrap;
    justify-content: flex-start;
    padding-left: 10px;
    font-size: 32px;
  `

  const StyledButton = styled.button`
    background-color: #aed581;
    border: none;
    color: #f1f8e9;
    padding: 7px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    margin: 10px;
    
    &:hover{
      cursor: pointer;
    } 
  `

  return(
    <div>
      <StyledButton 
        onClick = { handleClickOpen }
      >
        More Info
      </StyledButton>

      <Dialog
        open={open}
        onClose={handleClose}
        aria-labelledby="alert-dialog-title"
        aria-describedby="alert-dialog-description"
      >
        <DialogTitle id="alert-dialog-title">
          <Row>
            <CoverColumn>
              <BookCover src={props.bookCover} alt="bookcover"/>
            </CoverColumn>
            <TitleColumn>
              { props.book.title } <br />
              By {props.authors}
            </TitleColumn>
          </Row>
        </DialogTitle>
        <DialogContent>
          <DialogContentText>
            {props.book.description}
          </DialogContentText>
          <DialogContentText>
            {console.log(props.extraInfo)}
            Ratings: {props.book.ratingsCount} <br />
            Average Rating: {props.book.averageRating} <br />
            Categories: {props.book.categories} <br />
            Language: {props.book.language} <br />
            Page Count: {props.book.pageCount} <br />
            <button>
              <a href={props.book.infoLink}  target="_blank" rel="noopener noreferrer" >Even More Info</a> <br />
            </button>
          </DialogContentText>
        </DialogContent>
      </Dialog>
    </div>
  )
}

export default DialogComponent