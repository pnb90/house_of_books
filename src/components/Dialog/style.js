import styled from 'styled-components'

export const BookCover = styled.img`
  height: 266px;
  width: 200px;
  border: 1px solid grey;
  padding: 5px;    
`

export const Row = styled.div`
  display: flex;
  align-items: center;
`

export const CoverColumn = styled.div`
  flex: 25%;
`

export const TitleColumn = styled.div`
  flex: 75%;
  flex-wrap: wrap;
  justify-content: flex-start;
  padding-left: 10px;
  font-size: 32px;
`

export const StyledButton = styled.button`
  background-color: #aed581;
  border: none;
  color: #f1f8e9;
  padding: 7px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  margin: 10px;
  
  &:hover{
    cursor: pointer;
  } 
`

export const StyledListItem = styled.li`
  list-style-type: none;
  margin: 1rem;
`

export const StyledA = styled.a`
  background-color: #aed581;
  padding: 0.5rem;
  text-align: center;
  text-decoration: none;

  &:hover{
    cursor: pointer;
  } 
`