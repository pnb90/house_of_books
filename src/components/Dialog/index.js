import React, {useState} from 'react'
import { Dialog, DialogContent, DialogContentText, DialogTitle } from '@material-ui/core'
import { BookCover, Row, CoverColumn, TitleColumn, StyledButton, StyledListItem, StyledA } from './style' 
import PropTypes from 'prop-types'

/**
 * DialogComponent
 * @method 
 * 
 * @summary Renders a dialog modal for each individual book. 
 * 
 * @param {*} props - contains information about individual book from results of GoogleBooks API search. Includes book object, title, authors, bookcover, publisher, and description.
 */ 
function DialogComponent(props) {
  
  const [open, setOpen] = useState(false)

  /**
   * Opens the dialog component
   * 
   * Material UI Dialog component takes in props of open=boolean. While true, the dialog is open, and vice versa. handleOpen sets the open props to true upon clicking the button, while handleClose sets the open props to false.
   */

  function handleOpen() {
    setOpen(true)
  }

  function handleClose() {
    setOpen(false)
  }

  return(
    <div
      data-testid = "dialog testID"
    >
      <StyledButton onClick = { handleOpen }> More Info </StyledButton>

      <Dialog
        data-testid = "materialDialog testID"
        open={open}
        onClose={handleClose}
        aria-labelledby="alert-dialog-title"
        aria-describedby="alert-dialog-description"
      >
        <DialogTitle id="alert-dialog-title">
          <Row>
            <CoverColumn>
              <BookCover src={ props.bookCover } alt={ props.title + "'s book cover"} />
            </CoverColumn>
            <TitleColumn>
              <StyledListItem> { props.title }  </StyledListItem> 
              <StyledListItem> By { props.authors } </StyledListItem>
            </TitleColumn>
          </Row>
        </DialogTitle>

        <DialogContent>
          <DialogContentText component={'div'}>
            <StyledListItem> { props.description } </StyledListItem>  
            <StyledListItem> 
              <StyledA href={ props.externalLink }  target="_blank" rel="noopener noreferrer">
                Even More Info
              </StyledA> 
            </StyledListItem>  
          </DialogContentText>
        </DialogContent>

      </Dialog>
    </div>
  )
}

DialogComponent.propTypes = {
  book: PropTypes.object,
  bookCover: PropTypes.string,
  authors: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array
    ]),
  publisher: PropTypes.string
  
}

export default DialogComponent