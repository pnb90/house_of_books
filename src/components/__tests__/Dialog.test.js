import React from 'react';
import { mount, shallow, render } from '../../enzyme';
import Dialog from '../Dialog';

describe('Basic test', () => {
  it('basic test', () => {
    expect(2+2).toBe(4)
  })
})

// describe('Renders Dialog', () => {
//    it('renders without crashing', () => {
//       shallow(<Dialog />);
//     });
// });
