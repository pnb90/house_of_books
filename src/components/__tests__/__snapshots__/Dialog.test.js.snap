// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog component renders as expected 1`] = `
<DocumentFragment>
  <div
    data-testid="dialog testID"
  >
    <button
      class="sc-ifAKCX iFmML"
    >
       More Info 
    </button>
  </div>
</DocumentFragment>
`;
